spec/acceptance/letsencrypt_plugin_dns_cloudflare_spec.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'spec_helper_acceptance'
+
+describe 'letsencrypt::plugin::dns_cloudflare' do
+  context 'with defaults values' do
+    pp = <<-PUPPET
+      class { 'letsencrypt' :
+        email  => 'letsregister@example.com',
+        config => {
+          'server' => 'https://acme-staging-v02.api.letsencrypt.org/directory',
+        },
+      }
+      class { 'letsencrypt::plugin::dns_cloudflare':
+        api_token => 'dummy-cloudflare-api-token',
+      }
+    PUPPET
+
+    it 'installs letsencrypt and dns cloudflare plugin without error' do
+      apply_manifest(pp, catch_failures: true)
+    end
+
+    it 'installs letsencrypt and dns cloudflare idempotently' do
+      apply_manifest(pp, catch_changes: true)
+    end
+  end
```

Comment:
We have a shared example to make this shorter. See https://github.com/voxpupuli/voxpupuli-acceptance#an-idempotent-resource if you want to use this.

Reply:
Done, any hint about the failed tests on CentOS7? I can't find any issues why this happens.

Reply:
Seems it can't find `python2-certbot-dns-cloudflare` but it exists under EPEL.
Probably this is an ordering issue.

I see: `Notice: /Stage[main]/Epel/Epel::Rpm_gpg_key[EPEL-7]/Exec[import-EPEL-7]/returns: executed successfully`

after trying to install `python2-certbot-dns-cloudflare`

Reply:
I think for EL7 you somehow need to make sure EPEL is pulled in. Looks like the other DNS plugins have `require letsencrypt` which takes care of it:
https://github.com/voxpupuli/puppet-letsencrypt/blob/291f4b218230cf24477b74122c07703d6d119ce5/manifests/plugin/dns_rfc2136.pp#L27

Reply:
Okay maybe i will add a Class dependency inside the acceptance test would be better:
```
Class['letsencrypt'] -> Class['letsencrypt::plugin::dns_cloudflare']
```
Because if i add `require letsencrypt` i will run into a dependency loop if i use the defined resource `letsencrypt::certonly`... maybe on the other plugins this will fail too and i assume no-one use this plugins currently..

Reply:
Perhaps it's then sufficient to `require letsencrypt::install`?

Reply:
Done... tried and works for me as well...